src/platform/mhv/secondary-nav/components/MhvSecondaryNavMenu.jsx

```diff
@@ -13,6 +14,84 @@ import MhvSecondaryNavItem from './MhvSecondaryNavItem';
+  const [isLargeText, setIsLargeText] = useState(false);
+
+  /**
+   * Function to check the font size and determine if the user has custom browser large font settings.
+   * This is wrapped in useCallback to optimize performance by preventing unnecessary re-renders
+   */
+  const checkFontSize = useCallback(() => {
+    try {
+      // Get the root font size (set in HTML or body)
+      const rootFontSize = parseFloat(
+        getComputedStyle(document.documentElement).fontSize,
+      );
+
+      // Create a test div to measure actual rendered text size
+      const testDiv = document.createElement('div');
+      testDiv.style.fontSize = '1rem'; // Set font size to 1rem for relative measurement
+      testDiv.style.padding = '0';
+      testDiv.style.position = 'absolute';
+      testDiv.style.left = '-9999px'; // Position off-screen
+      testDiv.style.lineHeight = 'normal';
+      testDiv.textContent = 'M'; // Use 'M' as it's typically the widest character in most fonts
+
+      // Append the test div to the body, measure it, then remove it
+      document.body.appendChild(testDiv);
+      const actualSize = testDiv.clientHeight;
+      document.body.removeChild(testDiv);
+
+      // Consider text "large" if root font size > 16px or actual rendered size > 20px
+      const newIsLargeText = rootFontSize > 16 || actualSize > 20;
+
+      setIsLargeText(newIsLargeText);
+    } catch (error) {
+      console.log('Error checking font size:', error.message);
+    }
+  }, []);
+
+  // Effect to set up font size checking and window resize handling
+  useEffect(
+    () => {
+      // Function to check and update font size, wrapped in setTimeout for browser consistency
+      const checkAndUpdateFontSize = () => {
+        try {
+          setTimeout(checkFontSize, 0);
+        } catch (error) {
+          console.log('Error in checkAndUpdateFontSize:', error.message);
+        }
+      };
+
+      // Initial check
+      checkAndUpdateFontSize();
+
+      let resizeObserver;
+      try {
+        // Set up ResizeObserver to detect changes in document element size
+        resizeObserver = new ResizeObserver(checkAndUpdateFontSize);
+        resizeObserver.observe(document.documentElement);
+      } catch (error) {
+        console.log('Error setting up ResizeObserver:', error.message);
+      }
+
+      // Add window resize event listener as a fallback for browsers not supporting ResizeObserver
+      window.addEventListener('resize', checkAndUpdateFontSize);
+
+      // Cleanup function to remove listeners and disconnect observer
+      return () => {
+        try {
+          if (resizeObserver) {
+            resizeObserver.disconnect();
+          }
+          window.removeEventListener('resize', checkAndUpdateFontSize);
+        } catch (error) {
+          console.log('Error in cleanup function:', error.message);
+        }
+      };
+    },
+    [checkFontSize],
+  ); // Re-run effect if checkFontSize changes
```

Comment:
Would it be possible to extract this functionality into a function/package and add tests?

Reply:
> Would it be possible to extract this functionality into a function/package and add tests?

Sorry. To clarify, I'm stating this as a possible next step for this effort, and not requesting that you @jzucadi specifically should do this, unless you want to.

Reply:
@radavis you are 100% right, this should be all extracted out of the file itself and include some tests. I did overlook this, let me get moving on that. Thank you for pointing this out!

Reply:
@radavis heads up, I was not able to get to this.